ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java

```diff
@@ -657,22 +678,103 @@ public void shouldInsertInto() throws Exception {
   }
 
   @Test
-  public void shouldHandleErrorResponseFromInsertInto() {
+  public void shouldHandlePullQueryTableException() throws Exception {
```

Comment:
Why is this test being added? Testing the specific error (query against a table this is not query-able) doesn't make sense in this class since it's not specific to the client. Regarding client behavior for this error, this test file already contains `shouldHandleErrorResponseFromExecuteQuery()` and `shouldHandleErrorResponseFromStreamQuery()`.

Reply:
I'm trying to cover more error cases. This is client behavior triggering exception by query non-queryable table. `shouldHandleErrorResponseFromExecuteQuery` and `shouldHandleErrorResponseFromStreamQuery` seem cover other behavior like query non-exist table. I feel query non-queryable table is also one of the cases. If you feel this is not needed, I can remove this

Reply:
As mentioned above, whether the table is queryable or not is not relevant to the client. For client test coverage, we should simply check that the client properly propagates errors from the server to the user. This is not the right place to try exercising all possible server-side errors. That belongs in the server unit/integration tests instead.